container/test-impl-base/src/main/java/org/jboss/arquillian/container/test/impl/RunModeUtils.java

```diff
@@ -47,31 +49,64 @@ private RunModeUtils() { }
     * @param testMethod
     * @return
     */
-   public static boolean isRunAsClient(Deployment deployment, Class<?> testClass, Method testMethod)
+   public static boolean isRunAsClient(Deployment deployment, TestClass testClass, Method testMethod)
    {
-      boolean runMethodAsClient = testMethod.isAnnotationPresent(RunAsClient.class);
-      boolean runClassAsClient = testClass.isAnnotationPresent(RunAsClient.class);
-
       boolean runAsClient = true;
-      if(deployment != null)
+      if (deployment != null)
       {
-         runAsClient =  deployment.getDescription().testable() ? false:true;
-         runAsClient =  deployment.isDeployed() ? runAsClient:true;
+         runAsClient = deployment.getDescription().testable() ? false : true;
+         runAsClient = deployment.isDeployed() ? runAsClient : true;
```

Comment:
Yes and No.

So if No deployment context is given via @OperatesOnDeployment, and there is only 1 deployment, then that deployment is the current default context (if i remember correctly). So if you e.g. have a Deployment in manual mode, and want to test to deploy it, we can't run you in container because it's not deployed yet, but we can't fail either because it's intended not to be deployed yet.